### ISSUE TITLE
ConcurrentModificationException

### ISSUE THREAD

**Issue by maxencehenneron** [JRakNet/JRakNet#70]
The recoveryQueue in RakNetSession.java isn't synchronized, causing such exceptions:

java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1466)
	at net.marfgamer.jraknet.session.RakNetSession.update(RakNetSession.java:837)
	at net.marfgamer.jraknet.server.RakNetServer.start(RakNetServer.java:803)
	at net.marfgamer.jraknet.server.RakNetServer$1.run(RakNetServer.java:837)

**Comment by whirvis** [comment]
Thanks for reporting this, I'll get onto fixing it right away!

**Comment by whirvis** [comment]
I have created concurrent types for `ByteMap`, `ShortMap`, `IntMap`, and `LongMap` with their respective names but with `Concurrent` prefixed behind them. They function the exact the same with their code completely unchanged except for the fact that they extend `ConcurrentHashMap` rather than just `HashMap` :) This issue should no longer happen. Sorry for taking so long to respond, I was gone from GitHub for much longer than I normally am so I did not see this until I finally logged back in.
